src/OptionsCompounderAave2.sol

```diff
@@ -22,92 +26,56 @@ error OptionsCompounder__FlashloanNotProfitable(
     uint256 fundsAvailable,
     uint256 fundsToPay
 );
+error OptionsCompounder__AssetNotEqualToPaymentToken();
+error OptionsCompounder__NotFinished();
+
+address constant BEETX_VAULT_OP = 0xBA12222222228d8Ba445958a75a0704d566BF2C8;
 
 /* Main contract */
-contract OptionsCompounder is FlashLoanReceiverBase, Ownable {
+// TODO: UUPSUpgradeable ?
+contract OptionsCompounder is IFlashLoanReceiver {
     /* Constants */
     uint8 constant MIN_NR_OF_FLASHLOAN_ASSETS = 1;
-    address constant BEETX_VAULT_OP =
-        0xBA12222222228d8Ba445958a75a0704d566BF2C8;
 
     /* Storages */
     IOptionsToken private optionToken;
+    ILendingPoolAddressesProvider private addressProvider;
     ISwapperSwaps private swapperSwaps;
-    address private vaultAddress; // BEETx vault not a strategy vault
-    address[] private strategies;
-    uint256 gain = 0;
-
-    address lastExecutor;
-    bool flashloanFinished = true;
-    mapping(address => uint256) senderToBalance; // storage variable which tracks amount to withdraw by address
-
-    modifier onlyStrategy() {
-        if (false == isStrategyAdded(msg.sender)) {
-            revert OptionsCompounder__NotAStrategy();
-        }
-        _;
-    }
+    ILendingPool private lendingPool;
+    address wantToken;
+    bool flashloanFinished;
+    uint256 gain = 0; // TODO: remove at the end
 
     /**
      * List of params which are initiated at the begining:
      * @param _optionToken - option token address which allows to redeem underlying token via operation "exercise"
      * @param _addressProvider - address lending pool address provider - necessary for flashloan operations
      * @param _reaperSwapper - address to contract allowing to swap tokens in easy way
-     * @param _strategies - strategies which can use this module
      * */
-    constructor(
+    // TODO: Add onlyInitializing !
+    function _OptionsCompounder_Init(
         address _optionToken,
         address _addressProvider,
         address _reaperSwapper,
-        address[] memory _strategies
-    )
-        FlashLoanReceiverBase(ILendingPoolAddressesProvider(_addressProvider))
-        Ownable(msg.sender)
-    {
+        address _wantToken
+    ) internal /* onlyInitializing */ {
         optionToken = IOptionsToken(_optionToken);
+        addressProvider = ILendingPoolAddressesProvider(_addressProvider);
+        lendingPool = ILendingPool(addressProvider.getLendingPool());
         swapperSwaps = ISwapperSwaps(_reaperSwapper);
```

Comment:
see below comments, the strategy itself has some of this info already, and we dont want to have different functions and conflicting access control schemas between this and smth thats gonna inherit from this.  there should be a way to abstract more stuff so that we dont have to duplicate all the functionality